### PR TITLE
docs: fix intro project names

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
It looks like we have changed the package naming rules such that a lot
of examples in our documentation are now producing warnings. This PR
fixes the `daml new` commands, though I believe we should rather change
the package name constraints back, as it looks like a gratuitous
backwards-compatibility break.

CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

**Review by cocreature — APPROVED** [review]
thanks! I’m a bit sad that we don’t have tests for this but I don’t know a way of testing for this that seems reasonable unfortunately.

File: docs/source/daml/intro/9_Dependencies.rst [hunk 1]
```diff
@@ -40,12 +40,12 @@ You'll get a whole lot of output. Under the header "DAR archive contains the fol
...
+The first file is something like ``intro7-1.0.0-887056cbb313b94ab9a6caf34f7fe4fbfe19cb0c861e50d1594c665567ab7625.dalf`` which is the actual compiled package for the project. ``*.dalf`` files contain Daml-LF, which is Daml's intermediate language. The file contents are a binary encoded protobuf message from the `daml-lf schema <https://github.com/digital-asset/daml/tree/main/daml-lf/archive>`_.  Daml-LF is evaluated on the Ledger by the Daml Engine, which is a JVM component that is part of tools like the IDE's Script runner, the Sandbox, or proper production ledgers. If Daml-LF is to Daml what Java Bytecode is to Java, the Daml Engine is to Daml what the JVM is to Java.
```
Comment (carrielaben-da) [inline docs/source/daml/intro/9_Dependencies.rst:43]:
```suggestion
The first file is something like ``intro7-1.0.0-887056cbb313b94ab9a6caf34f7fe4fbfe19cb0c861e50d1594c665567ab7625.dalf`` which is the actual compiled package for the project. ``*.dalf`` files contain Daml-LF, which is Daml's intermediate language. The file contents are a binary encoded protobuf message from the `daml-lf schema <https://github.com/digital-asset/daml/tree/main/daml-lf/archive>`_. Daml-LF is evaluated on the Ledger by the Daml Engine, which is a JVM component that is part of tools like the IDE's Script runner, the Sandbox, or proper production ledgers. If Daml-LF is to Daml what Java Bytecode is to Java, the Daml Engine is to Daml what the JVM is to Java.
```

**Review by carrielaben-da — APPROVED** [review]

File: docs/source/daml/intro/9_Dependencies.rst (hunk 1 above)
Comment (garyverhaegen-da) [reply docs/source/daml/intro/9_Dependencies.rst]: Thanks!

**Comment by garyverhaegen-da** [comment]
> thanks! I’m a bit sad that we don’t have tests for this but I don’t know a way of testing for this that seems reasonable unfortunately.

Very much agreed, especially since I just discovered these warnings have been emitted since 1.9.0.

**Pull request merged**